python/foglamp/services/core/api/filters.py

```diff
@@ -0,0 +1,245 @@
+            return web.HTTPBadRequest(reason=message)
+
+        # Sanity checks
+        if (plugin_name != loaded_plugin_name or loaded_plugin_type != 'filter'):
+             errorMessage = "Loaded plugin '" + loaded_plugin_name + \
+                            "', type '" + loaded_plugin_type + \
+                            "', doesn't match the specified one '" + \
+                            plugin_name + "', type 'filter'"
+             raise Exception(errorMessage)
+
+        #################################################
+        # Set string value for 'default' if type is JSON
+        # This is required by the configuration manager
+        ################################################# 
+        for key, value in plugin_config.items():
+            if (value['type'] == 'JSON'):
+                value['default'] = json.dumps(value['default'])
+
+        await cf_mgr.create_category(category_name=filter_name,
+                                     category_description=filter_desc,
+                                     category_value=plugin_config)
+
+        # Fetch the new created filter: get category items
+        category_info = await cf_mgr.get_category_all_items(category_name = filter_name)
+        if category_info is None:
+            message = "No such '%s' filter found" % filter_name
+            raise Exception(message)
+
+    except Exception as ex:
+        _LOGGER.error("Add filter, caught exception: " + str(ex))
+        raise web.HTTPInternalServerError(reason=str(ex))
+
+    # Success: return new filter content
+    return web.json_response({'filter': filter_name,
+                              'description': filter_desc,
+                              'value': category_info})
+
+"""
+    Add filter names to "filter" item in {service_name}
+
+    PUT /foglamp/filter/{service_name}/pipeline
+ 
+    'pipeline' is the array of filter category names to set
+    into 'filter' default/value properties
+
+    :Example: set 'pipeline' for service 'NorthReadings_to_PI'
+    curl -X PUT http://localhost:8081/foglamp/filter/NorthReadings_to_PI/pipeline -d 
+    '{
+        "pipeline": ["Scale10Filter", "Python_assetCodeFilter"],
+    }'
+
+    Configuration item 'filter' is added to {service_name}
+    or updated with the pipeline arryay
+
+    Returns the filter pipeline on success:
+    {"pipeline": ["Scale10Filter", "Python_assetCodeFilter"]} 
+
+    Query string parameters:
+    - append_filter=true|false       Default true
+    - allow_duplicates=true|false    Default true
+
+    NOTE: the method also adds the filters category names under
+    parent category {service_name}
+"""
+
+async def add_filters_pipeline(request):
+    try:
+        # Get inout data
+        data = await request.json()
+        # Get filters list
+        filter_list = data.get('pipeline')
+        # Get filter name
+        service_name = request.match_info.get('service_name', None)
```

Comment:
raise Badrequest if service_name or pipeline is None and check pipeline is list and cannot be an empty

Reply:
Pipeline can be empty: it0s the way we remove filters, instead of removing the whole "filter" item

Reply:
fixed: added the other checks